client/trino-client/src/main/java/io/trino/client/ResultRows.java

```diff
@@ -40,7 +40,12 @@ public Iterator<List<Object>> iterator()
         }
     };
 
-    static ResultRows fromIterableRows(Iterable<List<Object>> values)
+    static ResultRows wrapIterator(Iterator<List<Object>> values)
```

Comment:
this looks fishy. Do we still want `ResultRows` to be Iterable, given that you no longer can use it to iterate over contents multiple times (at least in some of the contexts)